chart/templates/ingress.yaml

```diff
@@ -36,6 +36,9 @@ metadata:
 {{ toYaml .Values.ingress.extraAnnotations | indent 4 }}
 {{- end }}
 spec:
+{{- if .Values.ingressClassName }}
+  ingressClassName: {{ .Values.ingressClassName }}
```

Comment:
Rather than using a top-level variable called `ingressClassName`, I think it would make more sense to nest this under the `ingress` field i.e. https://github.com/rancher/rancher/blob/release/v2.6/chart/values.yaml#L55-L70

Also, if we could figure out a good "default value" to populate into `values.yaml` (even if it is "blank") for documentation purposes, I think that would be great as well.

Reply:
Thanks for the quick feedback! 

Can definitely change `.Values.ingressClassName ` to `.Values.ingress.ingressClassName`. 

I'm not sure if a usable default value is possible since for example Nginx's Ingress uses `nginx` as its value and other Ingress classes probably also use their own name.  Would setting it to null or having it as a comment in the `values.yaml` file be better? Or both so it's a null value with a comment explaining its purpose above it?

Reply:
Keeping it as an empty string by default is my suggestion. I believe that the `if` check here would fail on an empty string and the `ingressClassName` would be left out.

Reply:
Sounds good, I'll verify it right now

Reply:
Ok so if the user's values.yaml file has
```
ingress:
  ingressClassName: nginx
```

then the `ingressClassName` gets injected as expected. If it is not included, the empty string does indeed cause the `if` check to fail and so it behaves as it did before.

I also added a comment, hope that's ok